services/core/java/com/android/server/mperspective/PerspectiveService.java

```diff
@@ -364,24 +391,25 @@ public void handleMessage(Message msg) {
                 case MSG_UPDATE_DESKTOP_STATE:
                     updateDesktopState();
                     break;
+                case MSG_DISABLE_DESKTOP_INTERACTIVE:
+                    updateDesktopInteractiveStateLocked(false);
             }
         }
     }
 
     private class MDisplayListener implements DisplayManager.DisplayListener {
-        // track the hdmi display id to check if it has been removed later
-        private int mHdmiDisplayId = -1;
+        // track the public presentation display id to check if it has been removed later
+        private int mPublicPresentationDisplayId = -1;
 
         @Override
         public void onDisplayAdded(int displayId) {
             Display display = mDisplayManager.getDisplay(displayId);
-            final boolean hdmiDisplayAdded = display.getType() == Display.TYPE_HDMI;
 
-            if (hdmiDisplayAdded) {
-                if (mHdmiDisplayId == -1) {
-                    mHdmiDisplayId = displayId;
-                    if (mHDMIAutoStart) {
-                        Log.i(TAG, "HDMI display added, scheduling desktop start...");
+            if (display.isPublicPresentation()) {
+                if (mPublicPresentationDisplayId == -1) {
+                    mPublicPresentationDisplayId = displayId;
+                    if (mPublicPresentationAutoStart) {
+                        Log.i(TAG, "Public presentation display added, scheduling desktop start...");
                         scheduleStartDesktopPerspective();
```

Comment:
I think we will need `scheduleEnableDesktopInteractive()` here similar to the logic in `onDisplayRemoved()` to handle the case where the desktop is already in `STATE_RUNNING` and needs to have the input enabled when a public presentation display is added.